patches/server/0860-Fix-player-flight-disabling-on-world-change.patch

```diff
@@ -0,0 +1,23 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: mxson <m@son.xxx>
+Date: Mon, 24 Jan 2022 18:54:45 -0800
+Subject: [PATCH] Fix player flight disabling on world change
+
+
+diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
+index 7b23535a680d2a8534dcb8dd87770f66fb982c13..3ee8d55ae71a8b598e131261a7f7de52106e3eab 100644
+--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
++++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
+@@ -1721,7 +1721,11 @@ public class ServerPlayer extends Player {
+ 
+     public void restoreFrom(ServerPlayer oldPlayer, boolean alive) {
+         this.textFilteringEnabled = oldPlayer.textFilteringEnabled;
+-        this.gameMode.setGameModeForPlayer(oldPlayer.gameMode.getGameModeForPlayer(), oldPlayer.gameMode.getPreviousGameModeForPlayer());
++
++        if (this.gameMode.getGameModeForPlayer() != oldPlayer.gameMode.getGameModeForPlayer()) { // Paper - only change the player's gamemode if it's different from their current one
```

Comment:
Paper comment is wrong.
You need a
// Paper start - (Some change comment here)
**(Code here)
(More code)**
// Paper end

Reply:
That being said, the setGameModeForPlayer shouldn't be moved down or moved with indentation, simply put the if statement around the current code. Adding a single line // Paper comment on the if statement start and end.

Reply:
Was using this as reference: https://github.com/PaperMC/Paper/pull/7344/commits/debbdcca320df1a8a117a9b442bc1d1a25aae2c0
Possibly that was the wrong commit to use, I've adjusted that now.